okta/http_client.py

```diff
@@ -42,12 +42,42 @@ async def send_request(self, request):
                 method=request["method"],
                 url=request["url"],
                 headers=self._default_headers,
-                data=request["data"] or {},
+                data={} if "data" not in request else request["data"],
                 timeout=self._timeout
             ) as response:
                 return (response.request_info,
                         response,
-                        await response.json(), None)
+                        await response.json(),
+                        None)
         except (aiohttp.ClientError, asyncio.exceptions.TimeoutError) as error:
             # Return error if arises
             return (None, None, None, error)
+
+    @staticmethod
+    def check_response_for_error(url, response_details, response_body):
+        """
+        Checks HTTP response for errors in the response body
+
+        Args:
+            url (str): URL of response
+            response_details (aiohttp.ClientResponse): Response object with
+            details response_body (json): Response body in JSON format
+
+        Returns:
+            dict, (OktaAPIError or HTTPError): Tuple of: dict repr of the
+            response (if no error), any error found
+        """
+        # Retrieve dictionary repr and response status code
+        dict_resp = json.loads(json.dumps(response_body))
+        status_code = response_details.status
+
+        # check if call was succesful
+        if 200 <= status_code <= 300:
```

Comment:
I think you mean `200 <= status_code <= 299`.... you don't want success on a 300 call